encoding/json/response.go

```diff
@@ -27,4 +27,7 @@ type Response struct {
 	Headers transport.Headers
 
 	// TODO Response context?
+
+	// TODO Allow users to specify that this JSON response contains an
```

Comment:
It wouldn't be a header. There is no concept of 200s or status codes in a JSON response. A JSON handler is in the shape,

    func MyHandler(Context, *json.Request, AnyJSONUnmarshalableType) (AnyJSONMarshalableType, *json.Response, error) { ... }

If someone is indeed using JSON APIs, they presumably use shapes similar to this in their response:

    {"response": {actual response goes here}}
    {"error": {error details go here}

For the second case, they would want the TChannel application error bit to be set. Since there's no standard way of returning application errors in JSON, it would have to be explicitly specified by the user that a particular response contains an application error. One way I can imagine of supporting this is by providing an `ApplicationError` field on `json.Response` which users can set as necessary.

---

(Rename references to `json.Response` with `json.ResMeta` as appropriate.)